core/space/services/services_sharing.go

```diff
@@ -222,3 +224,39 @@ func (s *Space) OpenSharedFile(ctx context.Context, hash, password, filename str
 		Location: decryptedFile.Name(),
 	}, nil
 }
+
+func (s *Space) ShareFilesViaPublicKey(ctx context.Context, bucketName string, paths []string, pubkeys []crypto.PubKey) error {
+	err := s.tc.ShareFilesViaPublicKey(ctx, bucketName, paths, pubkeys)
+	if err != nil {
+		return err
+	}
+	for _, pk := range pubkeys {
+		err := s.tc.SetMember(ctx, bucketName, pk, nil)
```

Comment:
Actually, this has to change because membership is now at the file level so doesn't make sense to add at bucket level